<!-- 
Files tuned for ethzasl_icp_mapping commit 52d78c96c2cf2f58a400c649c561dbfec95e41a4
and for libpointmatcher commit e048c4d1180b1de005af072bf137ea4d8cf2b826

"Long-term 3D map maintenance in dynamic environments"
-->

<launch>
	<!--
	<arg name="file_path" default="/home/fabio/Documents/ASL-Lidar/data/velodyne/2016-03-11/2016-03-11-12-11-53.bag"/>
	-->
	

	
	<!-- <arg name="rviz_config" /> -->
	
	
	<!--node pkg="rviz" type="rviz" name="visualization" output="screen" respawn="true" args="-d /home/xiaqing/.rviz/dynamic_slam.rviz" / -->
	
<!--
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 odom velodyne 100" />
-->

	<param name="/use_sim_time" value="false"/>
	<node name="laser_mapping_with_bag" type="laser_mapping_with_bag" pkg="laser_mapping" output="screen" >
		<param name="bagname" value="/media/dxq/D/data/bag/cherry/1205/14/2019-12-05-14-merge.bag"/>
		<param name="TASK" value="without_map"/>
		<param name="reSLAM_refKFID" value="9"/>
		<param name="BeginID" value="0"/>
		<param name="firstPose" value="$(find laser_mapping)/cfg/firstPose.txt" />
		<param name="Debug_ICP" value="False" />
		<param name="mapElevation" value="False" />
		
		<param name="bag_begin" value="150" />
		<param name="bag_duration" value="70" />
		
	
		<param name="subscribe_scan" value="false"/>
		<param name="subscribe_cloud" value="true"/>
		<remap from="scan" to="/scan"/>
		<param name="cloud_in" value="/lidar_center/velodyne_points"/>
		<param name="odom_msg_name" value="/thomas_velocity_controller/vel_out" />
		<param name="imu_msg_name" value="/imu_data" />
		<param name="image_msg_name" value="/camera/left/image_rect_color" />
		<param name="gps_msg_name" value="/ins/data" />

		<param name="MapSavingFile" value="/media/dxq/D/data/bag/cherry/1205/14/laser_map/map.xml"/>
		<param name="FramesSavingPath" value="/media/dxq/D/data/bag/cherry/1205/14/laser_map/frames/"/>
		
		<param name="MapReadingFile" value="$(find laser_mapping)/KeyFrame/map_smallspace.xml"/>
		<param name="FramesReadingPath" value="$(find laser_mapping)/KeyFrame/frames/"/>
		
		<param name="extrinsicTLC" value="$(find laser_mapping)/cfg/extrinsicTLC.txt"/>
		<param name="refinefile" value="$(find laser_mapping)/cfg/refine.txt"/>
		<param name="extrinsicTBC" value="$(find laser_mapping)/cfg/extrinsicTBC.txt"/>

		<param name="use_imu" value="false" />
		<param name="use_odom" value="false" />
		<param name="use_velo_Twist" value="false" />
		<param name="use_gps" value="true" />
		
		
		<param name="icpConfig" value="$(find laser_mapping)/cfg/icp_dynamic.yaml" />
		<param name="icpLoopConfig" value="$(find laser_mapping)/cfg/icp_dynamic_loop_closing.yaml" />
		<param name="inputFiltersConfig" value="$(find laser_mapping)/cfg/input_filters.yaml" />
		<param name="mapPostFiltersConfig" value="$(find laser_mapping)/cfg/map_post_filters.yaml" />
		<param name="maximumDistFiltersConfig" value="$(find laser_mapping)/cfg/maximumDistFiltering.yaml" />
		

		<param name="laser_frame" value="/velodyne"/>
		<param name="map_frame" value="/map" />
		<param name="imu_frame" value="/imu" />

		<param name="useROSLogger" value="false" />
		<param name="publishMapTf" value="true" />
		<param name="inputQueueSize" value="1" />
		<param name="tfRefreshPeriod" value="0.005" />
		<param name="minOverlap" value="0.2" /> 
		<param name="maxOverlapToMerge" value="0.99" /> 
		<param name="minReadingPointCount" value="700" /> 
		<param name="minMapPointCount" value="10000" />
		<param name="localizing" value="true" /> 
		<param name="mapping" value="true" />

		<param name="priorStatic" value="0.45"/>
		<param name="priorDyn" value="0.55"/>
		<param name="maxAngle" value="0.01"/>
		<param name="eps_a" value="0.1"/> 
		<param name="eps_d" value="0.1"/>
		<param name="alpha" value="0.99"/>
		<param name="beta" value="0.90"/>ss
		<param name="maxDyn" value="0.90"/>
		<param name="maxDistNewPoint" value="0.2"/>
		<param name="sensorMaxRange" value="100.0"/>
		<param name="halfRoadWidth" value="40"/>
		<param name="MaxLocalMappoints" value="3000000"/>
		<param name="KFthreOverlap" value="0.6"/>
		<param name="maxLocalMapLength" value="20"/>
		<param name="windows_width" value="300"/>
		<param name="delta_width" value="1"/>
		<param name="initial_time" value="5"/>
		<param name="th_velo_up" value="1.0"/>
		<param name="min_add_dist" value="2.5"/>
		<param name="maxAngleforKF" value="7.0"/>
		<param name="staticKept" value="0.55"/>
		<param name="robot_height" value="2"/>
	</node>

</launch>
